### ISSUE TITLE
[SQL] Show tables should sort the result alphabetically be default

### ISSUE THREAD

**Issue by mpsq** [questdb/questdb#353]
In the current version, `SHOW TABLES` does not return the result in a sorted manner. Adding `ORDER BY tableName` does not help.

Most DBs sort the output of `SHOW TABLES` alphabetically by default, I think it would make sense if we did the same.

**Comment by ChristoperHowell** [comment]
I'll grab this as a first issue, if that's okay :)

**Comment by jttmtv** [comment]
I would like to work on this issue, is that ok?

**Comment by jerrinot** [comment]
hello @jttmtv, this looks like a good entry point: https://github.com/questdb/questdb/blob/8442ddb97dc0be09e29dea40892ffdadc8b2705d/core/src/main/java/io/questdb/griffin/SqlCompiler.java#L2575

it seems the magic is happening here: https://github.com/questdb/questdb/blob/6eca37b8f27bb1d51efc6cd8e68199ef2bd57b82/core/src/main/java/io/questdb/griffin/engine/table/TableListRecordCursorFactory.java#L97-L110

and last but not least: The `ShowTablesTest` will help you to get something up and running quickly. 

It seems to be the current mechanism is just iterating over directories as a directory means a table.

**Comment by jerrinot** [comment]
fwiw: this is a possible workaround `select * from all_tables order by table;` or simply `all_tables order by table;`

**Comment by rajdeepdas2000** [comment]
I would like to work on this issue.

**Comment by stephenspol** [comment]
Has this issue been resolved? When querying `show tables`, I got all the tables in ascending order. I also ran the test @jttmtv crafted in #353 and got a passing test.

**Comment by puzpuzpuz** [comment]
@stephenspol I don't think this one is resolved. Could be a coincidence that you're getting records in-order.

**Comment by Belair34** [comment]
I was testing this and it seems like the feature has been implemented in the current master branch (6.5.3), however it was not implemented in whatever version the live demo on the website is using. It's notable that in the table preview in the top left the tables are listed alphabetically in two groups: those that start with capitals and those that start with lowercase, however `SHOW TABLES` works as expected. It definitely doesn't seem coincidental given that I was able to reproduce it a lot.
![image](https://user-images.githubusercontent.com/46794728/191025440-f580e0e1-3c27-483c-8946-c49b324043b7.png)

**Comment by AyushShuklagithub** [comment]
Hi @mpsq , can I comtribute in this please ?

**Comment by AyushShuklagithub** [comment]
public static void main(String[] args) throws IOException {
    SqlCompiler compiler = new SqlCompiler();
    TableListRecordCursorFactory factory = new TableListRecordCursorFactory(compiler.configuration.getFilesFacade(), compiler.configuration.getRoot());
    List<String> tables = new ArrayList<>();
    while (factory.hasNext()) {
        tables.add(factory.next().getName());
    }
    tables.sort(String::compareToIgnoreCase);
    for (String table : tables) {
        System.out.println(table);
    }
}

**Comment by Somya2010** [comment]
Can I contribute in this one ?

**Comment by puzpuzpuz** [comment]
@Somya2010 sure thing. Thanks for willing to contribute!

**Comment by dhananjay123-v** [comment]
> In the current version, `SHOW TABLES` does not return the result in a sorted manner. Adding `ORDER BY tableName` does not help. Most DBs sort the output of `SHOW TABLES` alphabetically by default, I think it would make sense if we did the same.
> 
> **Note:** This is not a UI-only change regarding results in the Web Console, but requires changes in how the SQL is executed in the backend.

Parses SHOW TABLES

SELECT table_name FROM information_schema.tables 
WHERE table_schema = 'public'
ORDER BY table_name;

  Implementation Suggestion 

def handle_show_tables(schema: str = 'public'):
    sql = f"""
        SELECT table_name 
        FROM information_schema.tables 
        WHERE table_schema = '{schema}' 
        ORDER BY table_name
    """
    return execute_sql(sql)

**Comment by Prasun693** [comment]
Change the backend implementation of SHOW TABLES so that it explicitly sorts the list of tables by name in ascending alphabetical order before returning the result.

Note: This is not just a UI concern (e.g., for display in a Web Console) — it requires changes in the backend SQL execution layer that processes the SHOW TABLES command, to ensure that the returned result set is already ordered.

**Comment by nmarasoiu** [comment]
Hi, i see the following options and asking for advice:

Option 1: Introduce an SQL rewrite layer in SqlParser that would rewrite the normalized/cleaned strings like "show tables" and "all_tables()" to "all_tables order by table"

Option 2: 
Intro: io.questdb.cairo.MetadataCache.MetadataCacheReaderImpl#snapshot refreshes the local cache from tableMap; but neither support key order like TreeMap would, they are just a Map implemention which is fine, is performant and what was needed so far

Option 2a: Arrays.sort type thing from the local cache until cursor is created or reinitialized/rewind
Option 2b: Use a data structure which has key ordering in cache such as TreeMap
